tune/tune.c

```diff
@@ -65,7 +65,7 @@ static off_t exfat_get_root_entry_offset(struct exfat_blk_dev *bd)
 static int exfat_get_volume_label(struct exfat_blk_dev *bd, off_t root_clu_off)
 {
 	struct exfat_dentry *vol_entry;
-	char volume_label[11];
+	char volume_label[VOLUME_LABEL_MAX_LEN];
```

Comment:
This is incorrect. You should not expect that current locale codeset is set fixed 8bit encoding (which is even not common anymore as people are using UTF-8). IIRC correct macro should be `VOLUME_LABEL_BUFFER_SIZE`, so it can store `VOLUME_LABEL_MAX_LEN` characters in current locale multibyte codeset.

Reply:
VOLUME_LABEL_BUFFER_SIZE is defined as VOLUME_LABEL_MAX_LEN*MB_LEN_MAX+1
That would actually change 11 to another value, maybe fix it in another pr?

Reply:
And is not it correct?

Reply:
VOLUME_LABEL_MAX_LEN is 11
VOLUME_LABEL_BUFFER_SIZE would then be 11*16+1 here

Reply:
As @hclee explained to me before 11 is the max length of the label, so it's correct

Reply:
But buffer with 11 bytes is not large enough to store zero termined UTF-8 representation of UTF-16 string with 11 Unicode code points.

Reply:
> 11 is the max length of the label

Yes, 11 is maximal number of Unicode code points in label. But if your current locale codeset encoding does not use one byte for one Unicode code point, then 11 byte length buffer is not enough. Plus you forget additional null term for C string.

Reply:
Then that's a bug in the implementation :)
Should be fixed, thanks for pointing out, leaving it to the others I guess.

Reply:
@luca020400 As pali said, Could you change the array size? You can refer to VOLUME_LABEL_BUFFER_SIZE macro in fsck.h